### ISSUE TITLE
Select FP precision at build-time

### ISSUE THREAD

**Issue by mauneyc-LANL** [lanl/singularity-eos#154]
There is sufficient infrastructure is in place to compile a library that stores reals of arbitrary size, but this capability is not yet available downstream.

- `ports-of-call` should control this, but as near as I can determine this isn't available as a provided build option in that code. (There is a compile option, but this isn't available in the cmake interface or exported to downstream codes)
- eye-balling some snippets there is still some "loose" FP typing, e.g. https://github.com/lanl/singularity-eos/blob/ffda4d1a0d3a63c51c365c5945db588ba97baafc/singularity-eos/eos/singularity_eos.cpp#L431 . If using an explicit FP type is unavoidable, it's worth asking if `singularity-eos` should be responsible for explicitly casting to it

**Comment by Yurlungur** [comment]
this should be pretty to resolve with some care. Is this needed in the near future, @mauneyc-LANL ?

**Comment by jhp-lanl** [comment]
I'd say that right now our top priority is ensuring that we have a reliable and accurate interface to Pagosa that meets the minimum needs we are capable of fulfilling right now without performance considerations. The next step is obviously performance as well as expanding capabilities of singularity to meet more needs.

Today's stand-up meeting should be a good place to discuss progress and figure out what's next. Another area of performance we need to target is utilizing the vector interface for EOSPAC in a more efficient way. That's a heavier lift though so I'm all in favor of selecting some low-hanging fruit first.

**Comment by Yurlungur** [comment]
I agree with @jhp-lanl here. That said, issue should obviously remain open, as it's something we need to tackle eventually.

**Comment by Yurlungur** [comment]
This would also be an opportunity to have cmake generate a file with definitions rather than generating huge list of defs on the compile line.

Precision also an issue.
